### DISCUSSION TITLE
Sharing a token across an org

### DISCUSSION THREAD

**Discussion by et** [community/community#27178] [Actions]
<p>We have published a private rubygem and I’d like for that gem to be added to a Gemfile in another private repository that is used by around 60+ devs. I don’t want those 60+ devs to have to generate an access token to download this rubygem.</p>
<p>Is there a way for use a shared token and commit that that our private repository?</p>
<p>As an example, <a href="https://sidekiq.org/products/pro.html" rel="nofollow noopener">sidekiq-pro</a> does this by simply adding a .bundle/config to the repository with the following contents:</p>
<p>—<br>
BUNDLE_GEMS <strong>CONTRIBSYS</strong> COM: “shared:token”</p>
<p>Totally understand that this is not the most secure since it requires manaully rotating the token if a developer leaves but it is that path that requires the least amount of friction.</p>

**Comment by bcomnes** [comment, marked as answer]
<p>The way we solved a similar problem was by creating a machine user account, adding it to our org, and then generating a personal access token from that account.  </p>
<p>Then, in actions that need to be able to access other private org repos, we would set that PAT as an action secret. Then inside your action, use that PAT to set up an authorized environment that can access org private repos.</p>
<p>In our case, we were authorizing Terraform, which references modules via https git, so all we needed to do was set up a netrc file prior to accessing other private repos.  This can be done with a simple run action or slightly less verbose action we wrote to do just this:</p>
<p><a href="https://github.com/little-core-labs/netrc-creds" rel="nofollow noopener">https://github.com/little-core-labs/netrc-creds</a></p>
<p>There is a GITHUB_TOKEN secret automatically added to all actions by default, though I found this token to only be scoped to access the repo the action is associated with, not the rest of the org.  It would be cool if github made the scopes on this token customizable to read the rest of the org, but thats not the case yet.</p>
<p>Hope that helps.</p>

**Comment by bcomnes** [comment]
<p>You can now set org secrets, available to all actions.  Very cool.  Still no terraform provider for that feature yet though.</p>

**Comment by bcomnes** [comment]
<p>Terraform GitHub provider supports org secrets now</p>
<p><a href="https://registry.terraform.io/providers/hashicorp/github/latest/docs/resources/actions_organization_secret" class="onebox" target="_blank" rel="noopener nofollow ugc">https://registry.terraform.io/providers/hashicorp/github/latest/docs/resources/actions_organization_secret</a></p>
